mage_ai/data_cleaner/column_types/column_type_detector.py

```diff
@@ -135,78 +137,76 @@ def infer_column_type(series, column_name, dtype, kwargs):
-            if matches / length >= DATETIME_MATCHES_THRESHOLD:
-                mdtype = ColumnType.DATETIME
+            lowercase_column_name = column_name.lower()
+            correct_phone_nums = clean_series.str.match(REGEX_PHONE_NUMBER).sum()
+            correct_zip_codes = clean_series.str.match(REGEX_ZIP_CODE).sum()
+            if correct_phone_nums / length >= NUMBER_TYPE_MATCHES_THRESHOLD and str_in_set(
+                lowercase_column_name, RESERVED_PHONE_NUMBER_WORDS
+            ):
+                return ColumnType.PHONE_NUMBER
+            elif correct_zip_codes / length >= NUMBER_TYPE_MATCHES_THRESHOLD and str_in_set(
+                lowercase_column_name, RESERVED_ZIP_CODE_WORDS
+            ):
+                return ColumnType.ZIP_CODE
             else:
-                correct_emails = clean_series.str.match(REGEX_EMAIL).sum()
-                correct_phone_nums = clean_series.str.match(REGEX_PHONE_NUMBER).sum()
-                correct_zip_codes = clean_series.str.match(REGEX_ZIP_CODE).sum()
-                lowercase_column_name = column_name.lower()
-                if correct_emails / length >= STRING_TYPE_MATCHES_THRESHOLD:
-                    mdtype = ColumnType.EMAIL
-                elif correct_phone_nums / length >= STRING_TYPE_MATCHES_THRESHOLD and str_in_set(
-                    lowercase_column_name, RESERVED_PHONE_NUMBER_WORDS
-                ):
-                    mdtype = ColumnType.PHONE_NUMBER
-                elif correct_zip_codes / length >= STRING_TYPE_MATCHES_THRESHOLD and str_in_set(
-                    lowercase_column_name, RESERVED_ZIP_CODE_WORDS
-                ):
-                    mdtype = ColumnType.ZIP_CODE
-                elif series_nunique == 2:
-                    mdtype = ColumnType.TRUE_OR_FALSE
-                else:
-                    if type(exact_dtype) is list:
-                        mdtype = ColumnType.TEXT
-                    elif clean_series_nunique / length >= 0.8:
-                        mdtype = ColumnType.TEXT
-                    else:
-                        word_count = clean_series.map(lambda x: len(str(x).split(' '))).max()
-                        if word_count > MAXIMUM_WORD_LENGTH_FOR_CATEGORY_FEATURES:
-                            mdtype = ColumnType.TEXT
-                        else:
-                            if clean_series_nunique <= kwargs.get(
-                                'category_cardinality_threshold', 255
-                            ):
-                                mdtype = ColumnType.CATEGORY
-                            else:
-                                mdtype = ColumnType.CATEGORY_HIGH_CARDINALITY
-    return mdtype
+                return ColumnType.NUMBER
+    else:
+        matches = clean_series.str.match(REGEX_DATETIME).sum()
+        if matches / length >= DATETIME_MATCHES_THRESHOLD:
+            return ColumnType.DATETIME
+        correct_emails = clean_series.str.match(REGEX_EMAIL).sum()
```

Comment:
Let's add a TODO to refactor some of the code here. There's some duplicate logic and we don't need to do regex matching for all the syntax for each column.

Reply:
Added TODO